File: Daily Challenge/C/Alternate Characters Matching 6.10.2025.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>
int main() {
    char s1[101], s2[101];
    scanf("%s%s", s1, s2);
    for (int i = 0; s1[i]; i++) s1[i] = tolower(s1[i]);
    for (int i = 0; s2[i]; i++) s2[i] = tolower(s2[i]);
    if (strcmp(s1, s2) == 0) { printf("NO
"); return 0; }
    int n = strlen(s1), even = 1, odd = 1;
    for (int i = 0; i < n; i += 2) if (s1[i] != s2[i]) even = 0;
    for (int i = 1; i < n; i += 2) if (s1[i] != s2[i]) odd = 0;
    if (even || odd) printf("YES
"); else printf("NO
");
    return 0;
}

File: Daily Challenge/Java/Alternate Characters Matching 6.10.2025.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s1 = sc.next().toLowerCase(), s2 = sc.next().toLowerCase();
        if (s1.equals(s2)) {
            System.out.println("NO");
            return;
        }
        boolean even = true, odd = true;
        for (int i = 0; i < s1.length(); i += 2)
            if (s1.charAt(i) != s2.charAt(i)) even = false;
        for (int i = 1; i < s1.length(); i += 2)
            if (s1.charAt(i) != s2.charAt(i)) odd = false;
        if (even || odd) System.out.println("YES");
        else System.out.println("NO");
    }
}